src/main/java/frc/team166/robot/subsystems/Manipulator.java

```diff
@@ -0,0 +1,246 @@
+
+	DoubleSolenoid manipulatorSolenoid = new DoubleSolenoid(RobotMap.Solenoids.MANIPULATOR_SOLENOID_A,
+			RobotMap.Solenoids.MANIPULATOR_SOLENOID_B);
+
+	AnalogInput irSensor = new AnalogInput(RobotMap.AnalogInputs.ir);
+
+	Encoder leftIntakeEncoder = new Encoder(RobotMap.Encoders.LEFT_ROLLER_A, RobotMap.Encoders.LEFT_ROLLER_B);
+	Encoder rightIntakeEncoder = new Encoder(RobotMap.Encoders.RIGHT_ROLLER_A, RobotMap.Encoders.RIGHT_ROLLER_B);
+
+	double ROLLER_RADIUS = 1.4375; //inches
+	double DIST_PER_PULSE_INTAKE = (((ROLLER_RADIUS * 2.0 * Math.PI) / 1024.0) / 12.0); //feet
+	double OPTIMAL_MOTOR_RATE = 6.81; //ft/s
+
+	double motorSpeed;
+
+	public Manipulator() {
+		addChild(rollers);
+		addChild(irSensor);
+		addChild(leftIntakeEncoder);
+		addChild(rightIntakeEncoder);
+
+		leftIntakeEncoder.setReverseDirection(true);
+		leftRoller.setInverted(true);
+
+		leftIntakeEncoder.setDistancePerPulse(DIST_PER_PULSE_INTAKE);
+		rightIntakeEncoder.setDistancePerPulse(DIST_PER_PULSE_INTAKE);
+	}
+
+	/**
+	 * Resets encoders
+	 * <p>
+	 * Calls the reset function on the left and right intake encoders
+	 */
+	public void resetEncoders() {
+		leftIntakeEncoder.reset();
+		rightIntakeEncoder.reset();
+	}
+
+	/**
+	 * Gets IR voltage
+	 * <p>
+	 * Returns the voltage value output from the IR sensor
+	 * 
+	 * @return The voltage from the IR sensor
+	 */
+	public double getIRDistance() {
+		return irSensor.getVoltage(); //Manipulate this data pending experimentation
+	}
+
+	public void openManipulator() {
+		manipulatorSolenoid.set(Value.kForward);
+		;
+	}
+
+	public void closeManipulator() {
+		manipulatorSolenoid.set(Value.kReverse);
+	}
+
+	public double getAvgEncoderRate() { //ft/s
+		double leftRate = leftIntakeEncoder.getRate();
+		double rightRate = rightIntakeEncoder.getRate();
+		return (leftRate + rightRate) / 2.0;
+	}
+
+	/**
+	 * Gets average encoder distance
+	 * <p>
+	 * Returns the average distance of the left and right encoders
+	 * 
+	 * @return The average distance of the two encoders
+	 */
+	public double getAvgEncoderDistance() { //ft
+		double leftDist = leftIntakeEncoder.getDistance();
+		double rightDist = rightIntakeEncoder.getDistance();
+		return (leftDist + rightDist) / 2.0;
+	}
+
+	/**
+	 * Sets motors to intake mode
+	 * <p>
+	 * Sets the motors to 4/5 power inward to take in a cube on the field
+	 */
+	public void setMotorsToIntake() {
+		rollers.set(0.8); //change once you find optimal motor speed
+	}
+
+	/**
+	 * Sets motors to discharge mode
+	 * <p>
+	 * Sets the motors to 4/5 power outward to eject a stored cube
+	 */
+	public void setMotorsToDischarge() {
+		rollers.set(-0.8); //change once you find optimal motor speed
+	}
+
+	//COMMANDS
+	public void initDefaultCommand() {
+		setDefaultCommand(new SubsystemCommand(this) {
+
+			@Override
+			protected boolean isFinished() {
+				return false;
+			}
+
+		});
+	}
+
+	public Command DropCube() {
+		return new SubsystemCommand(this) {
```

Comment:
We didn't talk about this in training, because it's a new thing, but try using an ActionCommand:

`return new ActionCommand("Drop Cube", this, this::openManipulator);`

It's just a tad more concise, and meant specifically for situations when there's one action that's supposed to be run during initialization, then have the command finish

Reply:
Do this for CloseManipulator() below.